enum.c

```diff
@@ -4173,31 +4187,32 @@ slicewhen_i(RB_BLOCK_CALL_FUNC_ARGLIST(yielder, enumerator))
 
 /*
  *  call-seq:
- *     enum.slice_when {|elt_before, elt_after| bool } -> an_enumerator
+ *    slice_when {|element, next_element| ... } -> enumerator
  *
- *  Creates an enumerator for each chunked elements.
- *  The beginnings of chunks are defined by the block.
+ *  The returned enumerator uses the block
+ *  to partition elements into arrays ("slices");
+ *  it calls the block with each element and its successor;
+ *  begins a new slice is begun if and only if the block returns a truthy value:
  *
- *  This method splits each chunk using adjacent elements,
- *  _elt_before_ and _elt_after_,
- *  in the receiver enumerator.
- *  This method split chunks between _elt_before_ and _elt_after_ where
- *  the block returns <code>true</code>.
+ *    a = [0, 1, 2, 4, 5, 6, 8, 9]
+ *    e = a.slice_when {|i, j| j != i + 1 }
+ *    e.each {|array| p array }
  *
- *  The block is called the length of the receiver enumerator minus one.
+ *  Output:
  *
- *  The result enumerator yields the chunked elements as an array.
- *  So +each+ method can be called as follows:
+ *    [0, 1, 2]
+ *    [4, 5, 6]
+ *    [8, 9]
  *
- *    enum.slice_when { |elt_before, elt_after| bool }.each { |ary| ... }
+ *  Calls the block with each element and its successor;
```

Comment:
I think the slice_when documentation can end before this line, since the remaining parts basically repeat the same information.

Reply:
To be more clear, I would remove lines 4206-4249.  They just provide more examples, but the examples do not result in a better understanding of the method's behavior, IMO.